sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs

```diff
@@ -24,31 +25,56 @@ public MessageToPocoConverter(JsonSerializerSettings jsonSerializerSettings)
         public TElement Convert(ServiceBusReceivedMessage message)
         {
             // 1. If ContentType is "application/json" or binding to a JObject deserialize as JSON
-            // 2. If ContentType is not "application/json" attempt to deserialize using Message.GetBody, which will handle cases like XML object serialization
+            // 2. If ContentType is not "application/json" and it is a Data body message, attempt to deserialize using DataContractSerializer, which will handle cases like XML object serialization
             // 3. If this deserialization fails, do a final attempt at JSON deserialization to catch cases where the content type might be incorrect
 
             if (message.ContentType == ContentTypes.ApplicationJson ||
```

Comment:
What happens if the POCO is `byte[]` or another form that implies "just give me the message body bytes"?

Reply:
Binding to a byte array would not result in the Poco converter being executed. Instead, it is handled here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToByteArrayConverter.cs